src/odemis/acq/stream/_base.py

```diff
@@ -1049,30 +1052,86 @@ def _onNewData(self, dataflow, data):
 
         self._shouldUpdateImage()
 
-    def getBoundingBox(self):
+    def getPixelCoordinates(self, p_pos):
+        """
+        Translate physical coordinates into data pixel coordinates
+        Args:
+            p_pos(tuple float, float): the position in physical coordinates
+
+        Returns(tuple int, int or None): the position in pixel coordinates or None if it's outside of the image
+
+        """
+        raw = self.raw[0]
+        im = self.image.value
+        if not raw and not im:
+            raise LookupError("Stream has no data and stream's image is not defined")
+        md = raw.metadata.copy()
+        mergeMetadata(md)
+        pxs = md.get(model.MD_PIXEL_SIZE, (1e-6, 1e-6))
+        rotation = md.get(model.MD_ROTATION, 0)
+        shear = md.get(model.MD_SHEAR, 0)
+        translation = md.get(model.MD_POS, (0, 0))
+        if rotation or shear:
+            size = raw.shape[-1], raw.shape[-2]
+            tform = AffineTransform(rotation=rotation, scale=pxs, shear=shear, translation=translation).inverse()
+            pixel_pos = tform(p_pos)
+            if 0 <= p_pos[0] <= size[0] and 0 <= p_pos[1] <= size[1]:
+                return int(pixel_pos[0] + size[0] / 2), - int(pixel_pos[1] - size[1] / 2)
```

Comment:
Please explicitly have a `else: return None`. That's the default in Python, but it's better to be explicit about what we return.